### PR TITLE
[BD-46] docs: refactoring Usage Insights

### PR DESCRIPTION
**Pull request by PKulkoRaccoonGang**
## Description

Issue: https://github.com/openedx/paragon/issues/2318

### Deploy Preview

[Usage Insights](https://deploy-preview-2267--paragon-openedx.netlify.app/insights)

## Merge Checklist

* [ ] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [ ] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [ ] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [x] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [ ] Were your changes tested in the `example` app?
* [ ] Is there adequate test coverage for your changes?
* [ ] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [ ] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [ ] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @PKulkoRaccoonGang!


When this pull request is ready, tag your edX technical lead.

File: www/src/utils/getParagonComponentsTypes.js [hunk 1]
```diff
@@ -8,6 +8,9 @@ const getParagonComponentsTypes = (components) => {
     const isContext = !!component.Consumer && !!component.Provider;
     let componentType;
     switch (true) {
+      case componentName.toLowerCase() === 'paragon':
+        componentType = 'Paragon';
+        break;
```
Comment (adamstankiewicz) [inline www/src/utils/getParagonComponentsTypes.js:13]: Interesting; didn't know we had a usage of `import * as paragon from '@edx/paragon';`!

File: www/utils/createPages.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+              components
+            }
+            slug
+          }
+        }
+      }
+    }
+  `);
+  if (result.errors) {
+    reporter.panicOnBuild('🚨  ERROR: Loading createPages query');
```
Comment (adamstankiewicz) [inline www/utils/createPages.js:38]: (lol at this function name `panicOnBuild`)

**Review by adamstankiewicz — APPROVED** [review]
Awesome job cleaning up these "Usage Insights" related files! Should be more maintainable / 
 understandable / approachable moving forward.

**Comment by openedx-webhooks** [comment]
@PKulkoRaccoonGang 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/paragon&entry.867055334=https://github.com/openedx/paragon/pull/2267&entry.1484655318=https://github.com/PKulkoRaccoonGang&entry.752974735=2023-05-05+06:57&entry.1917517419=2023-06-16+14:54&entry.2133058324=Yes) so we can improve your experience in the future.

File: www/gatsby-node.js [hunk 3]
```diff
@@ -4,193 +4,16 @@
...
-const css = require('css');
+const createPages = require('./utils/createPages');
+const onCreateNode = require('./utils/onCreateNode');
+const onCreateWebpackConfig = require('./utils/onCreateWebpackConfig');
+const createCssUtilityClassNodes = require('./utils/createCssUtilityClassNodes');
 
-const fs = require('fs');
-const { INSIGHTS_PAGES } = require('./src/config');
-const { getThemesSCSSVariables, processComponentSCSSVariables } = require('./theme-utils');
+exports.onCreateWebpackConfig = ({ actions }) => onCreateWebpackConfig(actions);
```
[the remark this answers is not in the document]
Comment (PKulkoRaccoonGang) [reply www/gatsby-node.js:12]: Thanks
Yes, I think it's a good choice for understanding Gatsby build flow

**Pull request merged**